csv-data-management/extract_data_from_csv.py

```diff
@@ -395,5 +412,30 @@ def handle_params(params,file_param):
                 raise ConnectorError('Invalid File IRI {0}'.format(value))
     except Exception as err:
         logger.info('handle_params(): Exception occurred {0}'.format(err))
-        raise ConnectorError('Requested resource could not be found with input type "{0}" and value "{1}"'.format
-                             (input_type, value.replace('/api/3/attachments/', '')))
\ No newline at end of file
+        raise ConnectorError('Requested resource could not be found with input type "{0}" and value "{1}"'.format(input_type, value.replace('/api/3/attachments/', '')))
+
+def _ds_filter(params,ds):
+    df = ds
+    if(params.get('filterInput')):
+            input_type = params.get('filterInput')
+            if input_type == 'Use Regex as Filter':
+                regex = True
+            elif input_type == 'Use \'isin\' as Filter':
+                regex = False
+
+    # Filter dataset 
+    if regex:
+        reg = params.get('filter')
+        columnName = params.get('filterColumnName')
+        df= df[df[columnName].str.match(reg)==True]
+    else:
+        filterValue = params.get('filter').split(",")
+        columnName = params.get('filterColumnName')
+        df= df[df[columnName].isin(filterValue)]
+
+    return df
+
+def _df_to_csv(df):
+    df.to_csv('/tmp/dataset.csv', encoding='utf-8', header='true',compression="zip")
```

Comment:
do you want to append a unique uuid to the name? will override each other if running in parallel playbooks.
also, when is this file cleaned up?

Reply:
did changes , will update PR

Reply:
Done